### ISSUE TITLE
Text is not well highlighted when using textprop in diff mode

### ISSUE THREAD

**Issue by rickhowe** [vim/vim#7392]
**Describe the bug**
Text is well highlighted when using `matchaddpos()` but not `textprop` in diff mode.

**To Reproduce**
I am checking to highlight some text in diff mode.
For example, the following 2 windows are now in diff mode with the default color scheme

![01](https://user-images.githubusercontent.com/3096934/100537479-5642b080-326c-11eb-8ee8-10aa1a577edb.png)

And execute below in the left window.
```
hi TestHL guibg=Blue guifg=Yellow gui=italic,nocombine

let hl = 'TestHL'
let col = 3
let len = 5

" Line 1: matchaddpos()
let ln = 1
call clearmatches()
call matchaddpos(hl, [[ln, col, len]])

" Line 2: prop_type_add() and prop_add()
let ln = 2
call prop_type_delete(hl)
call prop_type_add(hl, {'highlight': hl})
call prop_clear(ln)
call prop_add(ln, col, {'length': len, 'type': hl})
```
The, we can see as below.

![02](https://user-images.githubusercontent.com/3096934/100537564-f1d42100-326c-11eb-9d9a-1f6481bf89eb.png)

**Expected behavior**
Line 1:
`matchaddpos()` can highlight text as expected if 'gui' includes nocombine.
Line 2:
When highlighting with `prop_type_add()` and `prop_add()`,
'guifg' is ok, but 'guibg' stays unchanged and 'gui' looks combined.
I tried to add `{'combine': 0}` in `prop_type_add()` but not yet well highlighted.

I would expect the same behavior between `matchaddpos()` and `textprop`.

**Environment (please complete the following information):**
 - vim 8.2.2067
 - Windows 10

**Comment by brammool** [comment]
Yes, combining match highlighting with diff highlighting, the match highlighting has priority.
Combining a text property with diff highlighting, the diff highlighting has priority.

We could perhaps use the existing priority field of a text property: when the value is 100 or more then it overrules diff highlighting (and perhaps some others).  Would that work?

**Comment by rickhowe** [comment]
I tried to add the priority as below, but the result is same. not well highlighted.

`call prop_type_add(hl, {'highlight': hl, 'priority': 1000})`

Does that what you suggested?

**Comment by brammool** [comment]
I didn't say adding a priority would work, it's only an idea for how we could change the code to make it work.

**Comment by rickhowe** [comment]
I find it not yet implemented.  It is nice to use the priority of textprop to overwrite tie diff highlighting. And also nice for matchadd because it can not overwrite the background color and attributes of the diff highlighting.
